x-py.md

```diff
@@ -0,0 +1,392 @@
+---
+layout: default
+title: Building rustc with x.py; The Rust Forge
+---
+
+# What is `x.py`?
+
+`x.py` is the script used to orchestrate the tooling in the `rustc` repository.
+It is the script that can build docs, run tests, and compile `rustc`. It is the
+now preferred way to build `rustc` and it replaces the old makefiles from
+before. Below are the different ways to utilize `x.py` in order to effectively
+deal with the repo for various common tasks.
+
+# Table of Contents
+- [Building `rustc`](#building-rustc)
+  - [Build everything](#build-everything)
+  - [Build different stages](#build-different-stages)
+  - [Build specific components](#build-specific-components)
+  - [Other Flags](#other-flags)
+- [Testing rustc](#testing-rustc)
+  - [Run all tests](#run-all-tests)
+  - [Run specific tests](#run-specific-tests)
+  - [Other Flags](#other-flags-2)
+- [Benchmarking rustc](#benchmarking-rust)
+  - [Other Flags](#other-flags-3)
+- [Documenting rustc](#documenting-rustc)
+  - [Document everything](#document-everything)
+  - [Document specific components](#document-specific-components)
+  - [Document internal rustc items](#document-internal-rustc-items)
+  - [Other Flags](#other-flags-4)
+- [Cleaning out build directories]()
+  - [Other Flags](#other-flags-5)
+- [Build distribution artifacts]()
+  - [Other Flags](#other-flags-6)
+- [Install distribution artifacts]()
+  - [Other Flags](#other-flags-7)
+
+# Building `rustc`
+If you want to build the compiler to add new features you'll need to run one of
+the build commands. Note that the first time you build the compiler it will take
+a longer time to do so since you'll need to compile the entire LLVM library.
+However, subsequent builds won't be as long unless LLVM is updated.
+
+### Build everything
```

Comment:
Yeah this sort of depends on the definition of "everything" you'd like to have. This does not literally build all programs associated with the build process, but it on the other hand builds everything relevant to running a compiler. Maybe this can be titled "Building a compiler" ?